athena/utils/data_loader.py

```diff
@@ -0,0 +1,53 @@
+"""Module for loading the MNIST data"""
```

Comment:
data_loader could be a class with load_mnist_data and download_mnist_data methods. Hope you're into it. I would also consider kind of default file path for mnist_data (let's say in /bin/mnist_data directory). Then we could provide high-level function load_mnist_data without filename argument that knows where ta take it from itself and whether it was downloaded or not. It would be more convenient.